lib/c/Lib_PrintBuffer.c

```diff
@@ -23,7 +23,9 @@ void Lib_PrintBuffer_print_compare(uint32_t len, uint8_t* buffer1, uint8_t* buff
   printf("\n");
 }
 
-void Lib_PrintBuffer_print_compare_display(uint32_t len, uint8_t* buffer1, uint8_t* buffer2) {
+void Lib_PrintBuffer_print_compare_display(uint8_t t1, uint8_t t2, uint32_t len, void* buf1, void* buf2) {
```

Comment:
This whole function was a bit dirty to begin with and many test files that use it were using admits to coerce ibuffers into buffers and vice versa. I agree that these functions may need to be revisited. For now, I changed these to be polymorphic over the buffer type, since we want to use both gbuffers and buffers when testing, and this was the minimal change that allowed existing tests to work (I left the admits wherever I saw them.) 

I agree that doing buffer-type-polymorphic printing functions would be nice but I could not see how to make KreMLin generate a nice interface in those cases.

Reply:
I think the way to make these functions buffer-polymorphic is to make them take const pointers; since you can inject all three varieties of buffers into const pointers *without* resorting to unsafe casts, this is how you achieve a buffer-type-polymorphic printing function that makes kremlin generate a nice interface without void*.

```fstar
val print_compare_display (b1 b2: LowStar.ConstBuffer.const_buffer UInt8.t): unit

...
// if you need to print, say, a mut buffer and an immut:
print_compare_display (LowStar.ConstBuffer.of_buffer b1) (LowStar.ConstBuffer.of_buffer b2)
```

and you also get a nice signature in C:
```
// Lib_PrintBuffer.h
void Lib_PrintBuffer_print_compare_display(uint32_t len, const uint8_t *b1, const uint8_t *b2)
```